apis/iam/v1beta1/serviceaccount_types.go

```diff
@@ -0,0 +1,117 @@
+	// Optional. A user-specified name for the service account.
+	// Must be less than or equal to 100 UTF-8 bytes.
+	// +optional
+	DisplayName *string `json:"displayName,omitempty"`
+
+	// Optional. A user-specified opaque description of the service account.
+	// Must be less than or equal to 256 UTF-8 bytes.
+	// +optional
+	Description *string `json:"description,omitempty"`
+}
+
+// ServiceAccountObservation is used to show the observed state of the
+// ServiceAccount resource on GCP.
+type ServiceAccountObservation struct {
+	// The resource name of the service account in the following format:
+	// projects/{PROJECT_ID}/serviceAccounts/{ACCOUNT}.
+	// part of https://godoc.org/google.golang.org/genproto/googleapis/iam/admin/v1#ServiceAccount
+	// not to be confused with CreateServiceAccountRequest.Name aka ServiceAccountParameters.ProjectName
+	// +immutable
+	Name string `json:"name,omitempty"`
+
+	// The id of the project that owns the service account.
+	// Note: This field is used in responses only. Any value specified here
+	// in a request is ignored.
+	ProjectID string `json:"projectId,omitempty"`
+
+	//The unique and stable id of the service account.
+	//Note: This field is used in responses only.
+	// Any value specified here in a request is ignored.
+	UniqueID string `json:"uniqueId,omitempty"`
+
+	//The email address of the service account.
+	// Note: This field is used in responses only.
+	// Any value specified here in a request is ignored.
+	Email string `json:"email,omitempty"`
+
+	// The OAuth2 client id for the service account.
+	// This is used in conjunction with the OAuth2 clientconfig API to make
+	// three legged OAuth2 (3LO) flows to access the data of Google users.
+	// Note: This field is used in responses only.
+	// Any value specified here in a request is ignored.
+	Oauth2ClientID string `json:"oauth2ClientId,omitempty"`
+
+	// A bool indicate if the service account is disabled.
+	// The field is currently in alpha phase.
+	// Note: This field is used in responses only.
+	// Any value specified here in a request is ignored.
+	Disabled bool `json:"disabled,omitempty"`
+}
+
+// A ServiceAccountSpec defines the desired state of a
+// ServiceAccount.
+type ServiceAccountSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  ServiceAccountParameters `json:"forProvider"`
+}
+
+// A ServiceAccountStatus represents the observed state of a
+// ServiceAccount.
+type ServiceAccountStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+	AtProvider                     ServiceAccountObservation `json:"atProvider,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// A ServiceAccount is a managed resource that represents a Google IAM Service Account.
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="STATUS",type="string",JSONPath=".status.bindingPhase"
+// +kubebuilder:printcolumn:name="STATE",type="string",JSONPath=".status.atProvider.state"
```

Comment:
I don't think we have a `state` field in our observation here

Reply:
Also, these printer columns are becoming pretty difficult to maintain across all of our resources. I wonder if we could set up a custom linter to ensure the fields we provide are valid :thinking: anyway, ignore my musings and carry on :)

Reply:
my musing is that 2 of the pieces of information (type and path) would be implicit if these were defined as tags on the fields themselves, like `k8sprintcolumn:"STATUS"` 🤔 (the would also be parsable/inspectable by standard go tools vs being special kubebuilder tags)

Reply:
`bindingPhase` only matters when we have claims against this resource, right? Does it makes sense to leave it out for now?

Reply:
Hey look I put some actually useful columns in!
```
$ kubectl get serviceaccounts.iam.gcp.crossplane.io
NAME              DISPLAYNAME   EMAIL                                                             DISABLED
perfect-test-sa                 perfect-test-sa@kasey-gcp-provider-test.iam.gserviceaccount.com  
```

Reply:
@kasey yep `bindingPhase` will be empty so your are welcome to omit if you like. However, hopefully resources like this will be able to be bound in some form or fashion relatively soon :)